### PR TITLE
Add semantic tokens highlighting for step arguments

### PR DESCRIPTION
**Pull request by uarlouski**
Screenshot 2022-12-19 at 10 35 41

### CODE REVIEW

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/server/VividusStudioLanguageServer.java [hunk 1]
```diff
@@ -129,7 +131,12 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams params)
                     .collect(collectingAndThen(toList(), ExecuteCommandOptions::new));
             capabilities.setExecuteCommandProvider(commandOptions);
 
-            initResult.setCapabilities(capabilities);
+            SemanticTokensLegend legend = new SemanticTokensLegend(List.of(
+                "vividus-step-argument"
+            ), List.of());
```
Comment (valfirst) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/server/VividusStudioLanguageServer.java:136]: can be one line

File: vividus-studio-server/vividus-studio-plugin/src/test/java/org/vividus/studio/plugin/testdata/TestData.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+package org.vividus.studio.plugin.testdata;
+
+import java.util.List;
+
+import org.vividus.studio.plugin.model.Parameter;
+import org.vividus.studio.plugin.model.StepDefinition;
+
+public class TestData
```
Comment (valfirst) [inline vividus-studio-server/vividus-studio-plugin/src/test/java/org/vividus/studio/plugin/testdata/TestData.java:27]: personally I'm not a fan of such approach, it leads to maintenance issues, when test data should be updated for one class with unit tests and should be preserved for another one

**Review by valfirst — APPROVED** [review]

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/match/TokenMatcher.java [hunk 3]
```diff
@@ -94,11 +101,13 @@ private static Outcome matchToken(Token token, String input, MutableInt position
 
         currentPos = matchIndex;
 
+        int inputLength = input.length();
         for (int tokenPos = 1; tokenPos < tokenAsString.length(); tokenPos++)
         {
             int nextPos = currentPos + tokenPos;
-            if (nextPos > input.length() - 1)
+            if (nextPos > inputLength - 1)
```
Comment (ikalinin1) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/match/TokenMatcher.java:108]: Please refactor to do -1 once

File: vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/service/StepDefinitionResolver.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        if (stepWrapper.isEmpty())
+        {
+            return Stream.empty();
+        }
+        Step step = stepWrapper.get();
+
+        return groupedStepDefinitions.get().get(step.getType())
+                                           .stream()
+                                           .map(tryToResolve(step))
+                                           .flatMap(Optional::stream);
```
Comment (ikalinin1) [inline vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/service/StepDefinitionResolver.java:84]:
```suggestion
     
        return  stepWrapper.map(step -> groupedStepDefinitions.get().get(step.getType())
                                          .map(Collection::stream)
                                          .orEsleGet(Stream::empty)
                                           .map(tryToResolve(step))
                                           .flatMap(Optional::stream);
```

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**